lib/TDL_provider.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:template/task.dart';
+import 'package:flutter/cupertino.dart';
+
+class TaskProvider extends ChangeNotifier {
+  final List<Tasks> _name = [];
+  String _filterBy = 'all';
+  List<Tasks> get getName {
+    return _name;
+  }
+
+  String get filterBy => _filterBy;
+
+  void setFilterBy(String filterBy) {
+    _filterBy = filterBy;
+    notifyListeners();
+  }
```

Comment:
Nice! Ett alternativ är att använda en setter, e.g.
```suggestion
  void set filterBy (String filterBy) {
    _filterBy = filterBy;
    notifyListeners();
  }
```

Istället för att anberopa funktionen `setFilterBy` när du vill ändra värdet kan du nu bara sätta det, e.g.

```dart
Provider.of<TaskProvider>(context, listen: false).filterBy = 'all';
```

men vilketsom funkar lika bra!